### PR TITLE
MSEARCH-236 Authority record search - Identifiers search option

### PR DESCRIPTION
**Pull request by Shans-Kaluhin**
### Purpose/Overview:

The purpose of this story is to support Authority record search option by following identifiers:
 Control Number, Library of Congress Control Number, Other standard Identifier and System Control number.  No other identifiers will be searchable.

### Assumptions:

The story assumes that identifiers array contains the fields that are populated with:
* 001
* 010 $a and $z subfields
* 024 with data from all available subfields with the exception of $6 and $8
* 035 System Control Number $a and $z subfields

### Requirements/Scope:
 * Exact match is supported
 ** Search by "81048315" returns only the records that match the search value

 * Left and right hand truncation is supported:
 ** Search by "81048*" - matches records that start with "81048"
 ** Search by "*8315" - matches records that end with "8315"

### Acceptance criteria:
 * User can search by LCCN
 * Exact match returns only the record that matches the searched value
 * Left- and right hand truncations are supported

### Additional Information:
 Structure of LCCN: [https://www.loc.gov/marc/lccn_structure.html]

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -348,41 +348,43 @@ if it is defined but doesn't match.
...
+| `sftGeographicTerm`            | full-text | `sftGeographicTerm == "geographic name"`     | Matches authorities with `geographic name` sft geographic term      |
+| `saftGeographicTerm`           | full-text | `saftGeographicTerm == "geographic name"`    | Matches authorities with `geographic name` saft geographic term     |
+| `uniformTitle`                 |   term    | `uniformTitle == "an uniform title"`         | Matches authorities with `an uniform title` uniform title           |
+| `sftUniformTitle`              |   term    | `sftUniformTitle == "an uniform title"`      | Matches authorities with `an uniform title` sft uniform title       |
+| `saftUniformTitle`             |   term    | `saftUniformTitle == "an uniform title"`     | Matches authorities with `an uniform title` saft uniform title      |
+| `identifiers.value`            |   term    | `identifiers.value = "1023*"`                | Matches authorities with the given identifier value |
+| `identifiers.identifierTypeId` |   term    | `identifiers.identifierTypeId=="123" identifiers.value = "1023*"` | Matches authorities that have an identifier if type `123` with value `1023*`|
```
Comment (Shans-Kaluhin) [inline README.md:379]: Just added here `identifiers.value` and `identifiers.identifierTypeId`

**Review by MikhailFokanov — APPROVED** [review]
Looks good.

**Review by MikhailFokanov — APPROVED** [review]

**Review by OleksiiKuzminov — APPROVED** [review]

**Review by pfilippov-epam — CHANGES_REQUESTED** [review]
HI, looks good except for a few things
* there is no need to add `identifierTypeId` as a search option, I am not sure that someone will use it and it's missing in acceptance criteria
* story says that search by LCCN must be provided, but currently, it can be performed only using queries like `identifiers.value == lccn` , which is ok, but I think that it's assumed that search option must be `lccn == lccn` (it can be done using searchField functionality)

File: src/main/java/org/folio/search/service/setter/authority/AbstractIdentifierProcessor.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import lombok.RequiredArgsConstructor;
+import org.folio.search.domain.dto.Authority;
+import org.folio.search.domain.dto.AuthorityIdentifiers;
+import org.folio.search.integration.InstanceReferenceDataService;
+import org.folio.search.service.setter.FieldProcessor;
+
+@RequiredArgsConstructor
+public abstract class AbstractIdentifierProcessor implements FieldProcessor<Authority, Set<String>> {
+
+  private final InstanceReferenceDataService identifierTypeCache;
```
Comment (pfilippov-epam) [inline src/main/java/org/folio/search/service/setter/authority/AbstractIdentifierProcessor.java:17]:
```suggestion
  private final InstanceReferenceDataService referenceDataService;
```

File: src/main/java/org/folio/search/service/setter/authority/AbstractIdentifierProcessor.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.folio.search.domain.dto.Authority;
+import org.folio.search.domain.dto.AuthorityIdentifiers;
+import org.folio.search.integration.InstanceReferenceDataService;
+import org.folio.search.service.setter.FieldProcessor;
+
+@RequiredArgsConstructor
+public abstract class AbstractIdentifierProcessor implements FieldProcessor<Authority, Set<String>> {
```
Comment (pfilippov-epam) [inline src/main/java/org/folio/search/service/setter/authority/AbstractIdentifierProcessor.java:15]: Maybe there is a way to reuse AbstractIdentifierProcessor from Instance?

**Review by pfilippov-epam — CHANGES_REQUESTED** [review]

**Review by pfilippov-epam — APPROVED** [review]

**Review by OleksiiKuzminov — APPROVED** [review]

**Review by MikhailFokanov — APPROVED** [review]

**Pull request merged**